src/main/java/net/sf/jabref/gui/PreviewPanel.java

```diff
@@ -275,16 +275,27 @@ public void setLayout(Layout layout) {
     }
 
     public void setEntry(BibEntry newEntry) {
-        if(entry.isPresent() && (entry.get() != newEntry)) {
-            entry.ifPresent(e -> e.removePropertyChangeListener(this));
-            newEntry.addPropertyChangeListener(this);
+
+        if (entry.isPresent() && (entry.get() != newEntry)) {
+            entry.ifPresent(e -> e.unregisterListener(this.currentChangeFieldUpdateEvent));
+            this.currentChangeFieldUpdateEvent = new ChangeFieldUpdateEvent();
+            newEntry.registerListener(this.currentChangeFieldUpdateEvent);
         }
+
         entry = Optional.ofNullable(newEntry);
 
         updateLayout();
         update();
     }
 
+    private class ChangeFieldUpdateEvent {
```

Comment:
Now I can understand You. Well this is possible but is the PreviewPanel object not too big for passing? I don't know how guava manages this... However I saw the following comment (https://github.com/JabRef/jabref/blob/master/src/main/java/net/sf/jabref/gui/PreviewPanel.java#L322). Now I cannot reconstruct the case where `vetoableChange(...)` method is called here. Probably we don't need such a listener here at all.